packages/server/graphql/types/GitLabIntegration.ts

```diff
@@ -0,0 +1,51 @@
+import {GraphQLList, GraphQLNonNull, GraphQLObjectType} from 'graphql'
+import {GQLContext} from '../graphql'
+import isValid from '../isValid'
+import IntegrationProviderOAuth2 from './IntegrationProviderOAuth2'
+import TeamMemberIntegrationAuthOAuth2 from './TeamMemberIntegrationAuthOAuth2'
+
+const GitLabIntegration = new GraphQLObjectType<any, GQLContext>({
+  name: 'GitLabIntegration',
+  description: 'Gitlab integration data for a given team member',
+  fields: () => ({
+    auth: {
+      description: 'The OAuth2 Authorization for this team member',
+      type: TeamMemberIntegrationAuthOAuth2,
+      resolve: async ({teamId, userId}, _args, {dataLoader}) => {
+        return dataLoader
+          .get('teamMemberIntegrationAuths')
+          .load({service: 'gitlab', teamId, userId})
+      }
+    },
+    cloudProvider: {
+      description:
+        'The cloud provider the team member may choose to integrate with. Nullable based on env vars',
+      type: IntegrationProviderOAuth2,
+      resolve: async (_source, _args, {dataLoader}) => {
+        const [globalProvider] = await dataLoader
+          .get('sharedIntegrationProviders')
+          .load({service: 'gitlab', orgTeamIds: ['aGhostTeam'], teamIds: []})
+        return globalProvider
+      }
+    },
+    sharedProviders: {
+      description: 'The non-global providers shared with the team or organization',
+      type: new GraphQLNonNull(new GraphQLList(new GraphQLNonNull(IntegrationProviderOAuth2))),
+      resolve: async ({userId}, _args, {dataLoader}) => {
+        const teamMembers = await dataLoader.get('teamMembersByUserId').load(userId)
+        const teamIds = teamMembers.map(({teamId}) => teamId)
+        const orgIds = Array.from(new Set(teamMembers.map(({orgId}) => orgId)))
+
+        const orgTeams = (await dataLoader.get('teamsByOrgIds').loadMany(orgIds))
+          .flat()
+          .filter(isValid)
+        const orgTeamIds = orgTeams.map(({teamId}) => teamId)
+        return dataLoader
+          .get('sharedIntegrationProviders')
+          .load({service: 'gitlab', orgTeamIds, teamIds})
+      }
+    }
+    // The GitLab schema get injected here as 'api'
```

Comment:
is this valid? I don't see anything related

Reply:
rootSchema is where the magic happens.
Can always check `./schema.graphql` for the truth!